DMCompiler/DM/Builders/DMExpressionBuilder.cs

```diff
@@ -456,6 +503,7 @@ private DMExpression BuildConstant(DMASTExpressionConstant constant) {
             case DMASTConstantString constString: return new String(constant.Location, constString.Value);
             case DMASTConstantResource constResource: return new Resource(Compiler, constant.Location, constResource.Path);
             case DMASTConstantPath constPath: return BuildPath(constant.Location, constPath.Value.Path);
+            case DMASTModifiedType constModifiedPath: return BuildPath(constant.Location, constModifiedPath.Value.Path);
```

Comment:
`var/type = /datum{foo="bar"}` would just end up being equal to `var/type = /datum`? Should probably add an unimplemented warning for that.